src/test/java/utils/SeleniumActions.java

```diff
@@ -17,11 +21,37 @@ public static void click(By by, WebDriver driver) {
             } catch (ElementClickInterceptedException | StaleElementReferenceException e) {
                 System.out.println("Can't find it man " + e.getMessage());
                 try {
-                    Thread.sleep(10);
+                    Thread.sleep(3000);
                 } catch (InterruptedException ex) {
                     throw new RuntimeException(ex);
                 }
             }
         }
     }
+
+    public static void clickUntil(By toBeDisplayed, By toClick, WebDriver driver) {
+        int maxLoopCount = 0;
+        while (!driver.findElement(toBeDisplayed).isDisplayed() && maxLoopCount < 5) {
+            System.out.println("element is not displayed");
+            maxLoopCount++;
+            SeleniumActions.click(toClick, driver);
+            try {
+                if (driver.findElement(toBeDisplayed).isDisplayed()) {
+                    System.out.println("element is displayed");
+                    Thread.sleep(3000);
+                } else {
+                    System.out.println("element is not displayed");
+                    Thread.sleep(3000);
+
+                }
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    public static void wait(By by, WebDriver driver) {
```

Comment:
rename this. wait doesnt tell you what its waiting for.

Reply:
Done